src/parcsr_ls/par_mgr_setup.c

```diff
@@ -1179,78 +1177,104 @@ hypre_MGRSetup( void               *mgr_vdata,
          }
 #endif
       }
-
-      // Check user-prescribed F-solver
-      if (lev == 0 && (mgr_data -> fsolver_mode) == 0)
+    
+      if (Frelax_type[lev] == 2 || Frelax_type[lev] == 9 || Frelax_type[lev] == 99 ||
+               Frelax_type[lev] == 199)
       {
-         if (Frelax_type[lev] == 2)
+         // Check user-prescribed F-solver
+         if (lev == 0 && (mgr_data -> fsolver_mode) == 0)
          {
-            // Check that solver is setup
-            if (((hypre_ParAMGData*)aff_solver[lev])->A_array != NULL)
+            if (Frelax_type[lev] == 2)
             {
-               if (((hypre_ParAMGData*)aff_solver[lev])->A_array[0] != NULL)
+               // Check that solver is setup
+               if (((hypre_ParAMGData*)aff_solver[lev])->A_array != NULL)
                {
-                  // F-solver is already set up, only need to store A_ff_ptr
-                  A_ff_ptr = ((hypre_ParAMGData*)aff_solver[lev])->A_array[0];
-                  A_ff_array[lev] = A_ff_ptr;
+                  if (((hypre_ParAMGData*)aff_solver[lev])->A_array[0] != NULL)
+                  {
+                     // F-solver is already set up, only need to store A_ff_ptr
+                     A_ff_ptr = ((hypre_ParAMGData*)aff_solver[lev])->A_array[0];
+                     A_ff_array[lev] = A_ff_ptr;
+                  }
+                  else
+                  {
+                     if (my_id == 0)
```

Comment:
You should protect this with a `print_level` check or use the `hypre_error_w_msg` routine (I'm not sure I got the name exactly right).